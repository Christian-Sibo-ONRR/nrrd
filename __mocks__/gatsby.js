/* eslint-disable no-undef */
const React = require('react')
const gatsby = jest.requireActual('gatsby')

module.exports = {
  ...gatsby,
  graphql: ([queryString]) => getQueryName(queryString),
  Link: jest.fn().mockImplementation(({ to, ...rest }) =>
    React.createElement('a', {
      ...rest,
      href: to,
      'data-testId': 'BaseLink'
    })
  ),
  StaticQuery: jest.fn(),
  useStaticQuery: function (queryName) {
    const filePath = `./__mock_queries__/${ queryName }_results.json`

    try {
      return require(filePath)
    }
    catch (err) {
      console.warn(`Failed to require mock file for ${ queryName }`)
      return {}
    }
  },
}
function getQueryName (queryString) {
  const captureGroup = queryString.match(/query\s([^\s]+).*{/)
  if (captureGroup && captureGroup.length) {
    return captureGroup[1]
  }
}
