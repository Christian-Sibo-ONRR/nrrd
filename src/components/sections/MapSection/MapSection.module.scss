
.root {
    //margin:80px 0px;
}

.title {

}

.revenueTable {
}

.alignRight {

}

.container {
    display: flex;
 //   justify-content: space-between;
    flex-wrap: wrap;
}

.containerLeft {
    min-width: 280px;
  //  padding-right:30px;
     flex-basis: 70%;
     order:1;
}

.containerRight {
    min-width: 200px;
    flex-basis: 30%;
    order:2;
    //float: right;
    // min-height: 400px;/* bunt on proper height*/

    //    border: 1px solid;
}

.containerBottom {
    min-width:280px;
    flex-basis:100%;		    
    height: 600px;
    order:3;
    // min-height: 400px;/* bunt on proper height*/
    //height: 600px;/* bunt on proper height*/
    //    border: 1px solid;
}
