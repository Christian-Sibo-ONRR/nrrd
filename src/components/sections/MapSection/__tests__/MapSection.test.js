import React from 'react'
import renderer from 'react-test-renderer'
import MapSection from '../MapSection'
import {StaticQuery,useStaticQuery, graphql} from 'gatsby'

//let data={foo:"bar"};
let {data} = require('./graphql.json');

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render(data)
				    )
    useStaticQuery.mockImplementationOnce(() => {
	return data
    })
})

describe('MapSection', () => {
    it('renders correctly', () =>
       {
	   const tree = renderer
		 .create(<MapSection />)
		 .toJSON()
	   expect(tree).toMatchSnapshot()
  })
})
