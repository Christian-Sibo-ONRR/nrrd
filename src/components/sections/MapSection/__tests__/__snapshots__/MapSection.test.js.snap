// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapSection renders correctly 1`] = `
<section
  className="root"
>
  <div
    className="container"
  >
    <div
      className="containerLeft"
    >
      <h3 />
      <p />
    </div>
    <div
      className="containerRight"
    >
      <label
        htmlFor="location-selector"
      >
        Explore a state or offshore region:
      </label>
      <select
        className="select-dark-gray"
        defaultValue="Choose location"
        id="location-selector"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value="Choose location"
        >
          Choose location
        </option>
        <optgroup
          label="States"
        />
        <optgroup
          label="Offshore Regions"
        />
      </select>
    </div>
    <div
      className="containerBottom"
    >
      <div
        className="map"
      />
       
    </div>
  </div>
</section>
`;
